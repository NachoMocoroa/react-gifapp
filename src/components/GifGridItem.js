import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ({title, url}) => {

    return (
        <a 
            href={ url } 
            target="_blank" 
            rel="noopener noreferrer" 
            className="card-gif animate__animated animate__fadeIn">
            <div className="card-gif-wrapper">
                <img 
                    alt={ title } 
                    src={ url } 
                />
            </div>
            { title && <p className="card-gif-text">{ title }</p> }
        </a>
    );
};

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};

export default GifGridItem;
