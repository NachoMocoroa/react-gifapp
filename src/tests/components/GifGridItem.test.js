import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Test GifGridItem', () => {
    
    const title = 'Hi GIF';
    const url = 'https://localhost/image.gif';
    const wrapper = shallow(
        <GifGridItem 
            title={title} 
            url={url} />
    );

    test('should render GifGridItem component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should contain url in href attribute of a label', () => {
        const link = wrapper.find('a');
        expect(link.prop('href')).toBe(url);
    });

    test('should contain url/title in src/alt attribute of img label', () => {
        const image = wrapper.find('img');
        expect(image.prop('src')).toBe(url);
        expect(image.prop('alt')).toBe(title);
    });

    test('should contain title paragraph', () => {
        const titleParagraph = wrapper.find('p').text().trim();
        expect(titleParagraph).toBe(title);
    });

    test('should contain "animate__fadeIn" class in a label', () => {
        const link = wrapper.find('a');
        expect(link.hasClass('animate__fadeIn')).toBe(true);
    });

});
