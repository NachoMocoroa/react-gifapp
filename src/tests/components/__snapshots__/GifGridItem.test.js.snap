// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGridItem should render GifGridItem component 1`] = `
<a
  className="card-gif animate__animated animate__fadeIn"
  href="https://localhost/image.gif"
  rel="noopener noreferrer"
  target="_blank"
>
  <div
    className="card-gif-wrapper"
  >
    <img
      alt="Hi GIF"
      src="https://localhost/image.gif"
    />
  </div>
  <p
    className="card-gif-text"
  >
    Hi GIF
  </p>
</a>
`;
